Automattic/jetpack
# Subscribe modal: remove filters and enable for Jetpack sites

Follow up to earlier https://github.com/Automattic/jetpack/pull/33235 where I thought I removed the filters, but apparently there was still some left. 

<img width="1110" alt="Screenshot 2023-11-01 at 14 43 04" src="https://github.com/Automattic/jetpack/assets/87168/8dc4fd5a-c75d-4831-a8a5-30d91ba807b3">

<img width="1189" alt="image" src="https://github.com/Automattic/jetpack/assets/87168/5c66a52b-7064-4724-abe1-5c56d80dbb7b">

<img width="1454" alt="image" src="https://github.com/Automattic/jetpack/assets/87168/f0c7c259-1c58-4bba-b412-97294c1fcf69">

## Proposed changes:

* Remove filters that enable/disable subscribe modal
* Remove conditionally showing launchpad task for WP.com sites — the filter was always enabled for them anyway.

### Other information:


## Jetpack product discussion



## Does this pull request change what data or activity we track or use?



## Testing instructions:






* At settings -> Newsletter 
* See Subscribe modal setting
* Enabling/disabling works, the modal in the blog posts (try incognito) works
* The link to edit the templat under the setting works
